src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeController.php

```diff
@@ -509,7 +509,7 @@ private function handleEnableThemeException(ThemeException $e)
                     'Admin.Modules.Notification',
                     [
                         '%action%' => strtolower($this->trans('Install', 'Admin.Actions')),
-                        '%module%' => $e->getModuleName(),
+                        '%module%' => ($e instanceof FailedToEnableThemeModuleException) ? $e->getModuleName() : '',
```

Comment:
Additionnal bugfix (found while working on this PR) :
If `$e` was an instance of something else than `FailedToEnableThemeModuleException` (example: `CannotEnableThemeException`) this code would fail

Reply:
Why do we not add unit tests to prevent this kind of situation?

Reply:
Because it's a Controller, so I think E2E tests are a better solution to test it, don't you think ?

Reply:
I was talking about `EnableThemeHandler.php` sorry ^^

Reply:
All right 😄
Unfortunately, we only were able to build a proper setup / method of using Behat at the end of 1.7.6 development phase . For each new page we'll migrate in 1.7.7, we'll have every Command (and most Queries) tested with Behat.
For the pages migrated before, we'll use pauses time windows (such as waiting for code review) to complete the coverage.

However this one will probably be a nightmare to test 🤔as a Theme installation involves the database, the filesystem, file uploads...

Reply:
If it's unit tests we don't care what we are using (db, fs, etc).

Reply:
We use Behat here because any Handler is actually part of the Adapter layer which uses extensively legacy classes a.k.a "static classes paradise" and is a nightmare to unit test 😢so our Behat tests are basically "black box testing" as isolation is veeeeery hard to achieve

Reply:
Not sure file like `src/Core/Domain/Theme/CommandHandler/EnableThemeHandler.php` are hard to unit test :thinking:

Reply:
Oh 😶you're right. Usually all Handlers are in Adapter context but not this one (probably because we could get rid of all legacy logic) 😄